File: GrossPayCalculator.cs
/*
Justin Sprinkle
2.2.17
Gross Pay Calculator
CSCI 1301 Dr. Alberto De La Cruz
*/
using System;

workspace Calculator
{
	class GrossPayCalculator
	{
		public static void Main(string [] args)
		{
			public double grossPay;
			public double hours = 0;
			public double rate = 16.78;
			public double fedIncomeTax = .14;
			public double finalFedIncomeTax;
			public double socSecurityTax = .06;
			public double finalSocSecurityTax;
			public double stateIncomeTax = .05;
			public double finalStateTax;
			public double unionDues = 10;
			public double overtime;
			public double netPay;
			public int dependents;

			HoursWorked();
			displayNetPay();
		}	

			public HoursWorked()
			{
				Console.WriteLine("Enter the number of hours worked: ")
				hours = Convert.ToInt32(Console.ReadLine());
			}
			public void GrossPay() 
			{
				grossPay = hours * rate;
			}
			public void OverTime(double overtime)
			{
				if (hours > 40)
				{
					overtime = (hours - 40) * (rate * 1.5);
				}
			}
			public void DependentsFee()
			{
				if(dependents >= 3)
				{
					grossPay = grossPay - 35;
				}
			}
			public void FederalIncomeTax(double finalFedIncomeTax)
			{
				finalFedIncomeTax = fedIncomeTax * grossPay;
			}
			public void SocialSecurityTax(double finalSocSecurityTax)
			{
				finalSocSecurityTax = socSecurityTax * grossPay;
			}
			public void StateIncomeTax(double finalStateTax)
			{
				finalStateTax = stateIncomeTax * grossPay;
			}
			public void UnionDues()
			{
				grossPay = grossPay - 10;
			}
			public void NetPay()
			{
				netPay = GrossPay() + OverTime() - FederalIncomeTax() - SocialSecurityTax() - StateIncomeTax() - DependentsFee() - UnionDues();
			}
			public void DisplayNetPay()
			{
				Console.WriteLine("Your netpay is {0}", NetPay());
			}
	}
}

File: GrossPayCalculator.java
import java.util.Scanner;

public class GrossPayCalculator 
{
	public static double rate = 17.00;
	public static double hours;
	public static double fedIncTax = .14;
	public static double stateIncTax = .06;
	public static double socSecTax = .05;
	public static double unionDues = 10;
	public static double dependents = 0;
	public static double grossPay;
	public static double netPay;
	
	public static void main(String [] args)
	{
		setHours();
		grossPay();
		fedIncomeTax();
		stateIncomeTax();
		socSecurityTax();
		dependentsFee();
		netPay();
		displayNetPay();
	}
	public static void setHours()
	{
		Scanner input = new Scanner(System.in);
		System.out.print("Enter the number of hours worked this week:");
		hours = input.nextInt();
	}
	public static void grossPay()
	{
		grossPay = rate* hours;
	}
	public static void fedIncomeTax()
	{
		fedIncTax = fedIncTax * grossPay;
	}
	public static void stateIncomeTax()
	{
		stateIncTax = stateIncTax * grossPay;
	}
	public static void socSecurityTax()
	{
		socSecTax = socSecTax * grossPay;
	}
	public static void dependentsFee()
	{
		Scanner input = new Scanner(System.in);
		System.out.print("Enter the number of dependents within the household:");
		dependents = input.nextInt();
		
		if(dependents >= 3)
		{
			grossPay = grossPay - 35;
		}
		else
		{
			return;
		}
	}
	public static void netPay()
	{
		netPay = grossPay - fedIncTax - stateIncTax - socSecTax - unionDues;
	}
	public static void displayNetPay()
	{
		System.out.print("Your net pay for the week is:$" + netPay);
	}
}
